enhancements/storage/storage-performant-security-policy.md

```diff
@@ -0,0 +1,202 @@
+
+### Proof of concept
+
+We have already implemented a proof-of-concept https://github.com/openshift/kubernetes/pull/2311/files, which works and does the job. But this uses existing SCC admission hook.
+
+
+### Workflow Description
+
+Admins that want to set defaults of `fsGroupChangePolicy` and `selinuxChangePolicy` will set `storage.openshift.io/fsgroup-change-policy` or `storage.openshift.io/selinux-change-policy`
+labels in corresponding namespaces.
+
+This will result in setting these fields in all pods in affected namespace if these fields are *not* already set.
+
+### API Extensions
+
+Adds following labels as API conventions for defaulting of security policies:
+
+- `storage.openshift.io/fsgroup-change-policy`
+- `storage.openshift.io/selinux-change-policy`
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+It should not have effect on Hypershift deployments.
+
+#### Standalone Clusters
+
+It should not have effect on Standalone deployments.
+
+#### Single-node Deployments or MicroShift
+
+It should not have effect on Single-Node deployments.
+
+### Implementation Details/Notes/Constraints
+
+We will implement the necessary admission hooks in - https://github.com/openshift/apiserver-library-go/tree/master/pkg ,
+which will be feature-gated while in Tech Preview.
+
+### Risks and Mitigations
+
+This feature requires storage to be enabled for this to work. See Operational Aspect section for more details.
+
+### Drawbacks
+
+The admission hook implementation should be replaced by Mutating Admission Hook and hence it is kinda busy work while we wait for MAP implementation to go GA in upstream.
+
+## Alternatives (Not Implemented)
+
+None
+
+## Open Questions [optional]
+
+None
+
+## Test Plan
+
+We will add e2e tests that validate if labels are working as intended. These would be either part of CSO tests or `openshift/origin` tests.
+
+## Graduation Criteria
+
+The feature will be behind a feature-gate `StoragePerformantSecurityPolicy` during tech-preview.
+
+### Dev Preview -> Tech Preview
+
+N/A
+
+### Tech Preview -> GA
+
+During the initial development this feature will be behind a Openshift feature gate and will not be generally available. When we move this feature-gate to GA,
+it should be as simple as removal of the Openshift feature-gate.
+
+### Removing a deprecated feature
+
+N/A
+
+## Upgrade / Downgrade Strategy
+
+This should not have any impact on Upgrade and Downgrade strategy.
+
+## Version Skew Strategy
+
+Since this feature only applies to control-plane defautling, IMO we should be fine even older version of kubelets etc.
+
+## Operational Aspects of API Extensions
+
+There is obvious aspect of supporting forever a label on namespace as a API extension in Openshift. For most practical purposes
+this should work fine.
+
+When we remove the admission hook and switch to using MAP, CSO is the one that will install the MAP hook and hence storage *must* be
```

Comment:
Could CVO install the MAP? It is static right?

Reply:
Yes, we can have CVO install this. I can make an explicit note of it.